recipes-app/iot2050-efivarfs-helper/files/iot2050-efivarfs-helper.tmpl

```diff
@@ -0,0 +1,180 @@
+            print("{} the immutable flag of {}... ".format(action, filepath),
+                  end='')
+            cmd = 'chattr {} {}'.format(option, filepath)
+            result = subprocess.run(
+                cmd, shell=True, capture_output=True, check=True, text=True)
+            print("Done.")
+
+
+def do_prepare():
+    message_prepare_fail = "Fail to prepare the efivarfs for manipulating. \
+You may need to check the environment."
+
+    try:
+        mod_required = 'tee_stmm_efi'
+        print("Checking for the kernel module {}... ".format(mod_required),
+              end='')
+        cmd = 'lsmod'
+        result = subprocess.run(
+            cmd, shell=True, capture_output=True, check=True, text=True)
+        if len([i for i in result.stdout.split('\n')
+                if i.startswith("{} ".format(mod_required))]) == 1:
+            print("Done.")
+        else:
+            print("ERROR: {} is not loaded!".format(mod_required))
+            exit_on_failure(message_prepare_fail)
+    except subprocess.CalledProcessError as e:
+        dump_subprocess_error(e)
+        exit_on_failure(message_prepare_fail, code=e.returncode)
+
+    # tee-supplicant is used to serve the trusted applications running in
+    # optee-os for accessing RPMB partition.
+    try:
+        print("Checking for the service tee-supplicant... ", end='')
+        cmd = 'systemctl is-active tee-supplicant.service'
+        result = subprocess.run(
+            cmd, shell=True, capture_output=True, check=True, text=True)
+
+        print("Done.")
+    except subprocess.CalledProcessError as e:
+        print("ERROR: service is inactive!")
+        exit_on_failure(message_prepare_fail)
+
+    # efivarfs must be mounted with read & write option.
+    try:
+        print("Checking for efivarfs mounting... ", end='')
+        cmd = 'findmnt {}'.format(EFIVARFS_MOUNT_POINT)
+        result = subprocess.run(
+            cmd, shell=True, capture_output=True, check=True, text=True)
+        efivarfs_mount_error = 'not mounted.'
+        for line in result.stdout.split('\n'):
+            if line.startswith(EFIVARFS_MOUNT_POINT):
+                mount_options = line.strip().split(' ')[-1].split(',')
+                if 'rw' in mount_options and 'ro' not in mount_options:
+                    efivarfs_mount_error = ''
+                else:
+                    efivarfs_mount_error = 'not mounted as read & write.'
+
+        if efivarfs_mount_error != '':
+            print("ERROR:\tefivarfs is {}".format(efivarfs_mount_error))
+            exit_on_failure(message_prepare_fail)
+
+        print("Done.")
+    except subprocess.CalledProcessError as e:
+        dump_subprocess_error(e)
+        exit_on_failure(message_prepare_fail, code=e.returncode)
+
+    # efitools provides convenient commands to manipulate the UEFI authenticated
+    # variables.
+    try:
+        print("Checking for the efitools pre-installation... ", end='')
+        cmd = 'which efi-readvar'
```

Comment:
This may be better done with the included batteries: [shutil.which](https://docs.python.org/3.12/library/shutil.html#shutil.which)

Reply:
Indeed.